ledger/catchpointtracker.go

```diff
@@ -555,16 +555,19 @@ func (ct *catchpointTracker) commitRound(ctx context.Context, tx trackerdb.Trans
 		}
 
 		var trie *merkletrie.Trie
+		ct.catchpointsMu.Lock()
```

Comment:
Why not defer the Unlock? Is there a weird control path that I'm not seeing in the diff?